src/components/Asset/AssetActions/Compute/FormComputeDataset.tsx

```diff
@@ -119,6 +133,19 @@ export default function FormStartCompute({
     fetchAlgorithmAssetExtended()
   }, [values.algorithm, accountId, isConsumable])
 
+  useEffect(() => {
+    if (!values) return
+
+    const output = {}
+    Object.entries(values).forEach((value) => {
+      if (value[0] === 'algorithm') return
+      output[value[0]] = transformConsumerParametersForConsumption(
+        value[1] as FormConsumerParameter[]
+      )
+    })
+    setUserCustomParameters(output)
+  }, [setUserCustomParameters, values])
+
```

Comment:
issue(blocking): this needs comments or refactoring to increase readability. From just looking at the code it is hard to understand at first glance